### DISCUSSION TITLE
Using Websockets to fetch data and update Wave app

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/wave#2154] [Q&A]
Hi!I am trying to use Websockets to update a Wave dashboard, and have been struggling with it. Do you have any examples for it?
I tried using [autobahn Web sockets](https://github.com/crossbario/autobahn-python/tree/master) as well as [Websockets](https://github.com/python-websockets/websockets/tree/bf858ac9a6e047c50614d6990a12345e1f38802c).  What I'm trying to achieve is on the `btn_login` event handler within the `serve()` function, once authentication is complete, I am connecting to a web socket server url. The data will be received here which is then used to update my dashboard tables and graphs. 

I am getting an issue in threads while using `Autobahn`,. Any pointers would be much appreciated!

**Comment by mturoci** [comment]
> I am getting an issue in threads while using Autobahn

Can you elaborate? Is there any stacktrace? Are you sure it's not a problem with Autobahn?

The general design should be to connect to WS within background job (asyncio task should be enough) and then simply update the UI from there.

**Reply by n-srinidhi** [reply]
@mturoci Sorry for the late reply! Following up on this, Socket connection is happening, Problem is if I have two different client tabs(which are connected as two different web socket clients to my Web socket server), with two different user logins. Which means data updates are different for both. When data updates are being sent, it goes haywire and ends up propagating updates to all or the wrong tabs.
